app/routes.py

```diff
@@ -1,139 +1,165 @@
 from app import app, db
 from flask import jsonify, request
-from app.models import User, Post 
+from app.models import User, Post
 
 
 # Create, Update, Get, Delete one post from db
-@app.route('/post/<post_id>', methods =['GET', 'PUT', 'DELETE'])
+@app.route("/post/<post_id>", methods=["GET", "PUT", "DELETE"])
 def manage_post(post_id):
-    if request.method == 'DELETE':
-        post = Post.query.get(post_id)
+    if request.method == "DELETE":
+        post = Post.get_by_id(post_id)
         if post is None:
-            return jsonify({"title":"There is no post","msg":"No post deleted"}), 400
+            return jsonify({"title": "There is no post", "msg": "No post deleted"}), 400
         post.delete()
         return jsonify({"title": "Post deleted", "msg": "Post deleted from db"}), 204
-    
-    elif request.method == 'PUT':
+
+    elif request.method == "PUT":
         try:
             data = request.get_json(force=True)
         except Exception as error:
-            return jsonify({"title":"request failed", "msg": str(error)}), 400
+            return jsonify({"title": "request failed", "msg": str(error)}), 400
         body = data.get("body")
         if body is None:
-            return jsonify({"title":"There is no body","msg": "body is required"}), 400
-        post = Post.query.get(post_id)
+            return (
+                jsonify({"title": "There is no body", "msg": "body is required"}),
+                400,
+            )
+        post = Post.get_by_id(post_id)
         if post is None:
-            return jsonify({"title": "There is no post", "msg": "Post could not updated to db"}),400
+            return (
+                jsonify(
+                    {"title": "There is no post", "msg": "Post could not updated to db"}
+                ),
+                400,
+            )
         post.body = body
         try:
             post.save()
             return jsonify({"title": "Post updated", "msg": "Post updated to db"}), 200
         except Exception as error:
             db.session.rollback()
-            return jsonify({"title":"save failed", "msg": str(error)}), 400
-        
+            return jsonify({"title": "save failed", "msg": str(error)}), 400
 
     else:
-        if request.method == 'GET':
-            post = Post.query.get(post_id)
-        if post is None :
-            return jsonify({"title": "There is no post","msg":"No post found"}), 400
-        else :
-            return jsonify({ 
-            'body':post.body , 
-            'timestamp':post.timestamp,
-            'user_id':post.user_id
-        }), 200
-     
-
-@app.route('/post', methods =['POST'])
+        if request.method == "GET":
+            post = Post.get_by_id(post_id)
+        if post is None:
```

Comment:
If request.method is not GET the `post` will be undefined. You don't need line 44, instead
```python
else:
    post = Post.get_by_id(post_id)
    if post is None:
```